pkg/authz/authorizer.go

```diff
@@ -198,9 +200,21 @@ func (m Authorizer) Authenticate(w http.ResponseWriter, r *http.Request, upstrea
 			if m.AuthRedirectStatusCode > 0 {
 				redirOpts["auth_redirect_status_code"] = m.AuthRedirectStatusCode
 			}
-			//redirOpts["logger"] = m.logger
 
-			m.parseLoginHint(r, redirOpts)
+			if m.LoginHintEnabled {
+				queryHasLoginHint := r.URL.Query().Has("login_hint")
+				var loginHint = ""
+
+				if queryHasLoginHint {
```

Comment:
@PedroDBFlores , let's implement `switch` here. Also, let's not use `queryHasLoginHint` and `loginHint:

```
switch {
case r.URL.Query().Has("login_hint"):
  redirOpts["login_hint"] = r.URL.Query().Get("login_hint")
case !r.URL.Query().Has("login_hint") && usr != nil:
  # What if usr.Claims.Email (should be using usr.Authenticator.LoginHint instead; putting commit as I type) is empty?   Perhaps checking it might work better.
  redirOpts["login_hint"] =  usr.Authenticator.LoginHint  
}
```